xcdat/spatial.py

```diff
@@ -247,22 +247,23 @@ def get_weights(
             "Bounds",
             {"domain": xr.DataArray, "region": Optional[RegionAxisBounds]},
         )
-        axis_bounds: Dict[SpatialAxis, Bounds] = {
-            "X": {
+        axis_bounds: Dict[SpatialAxis, Bounds] = {}
+        if "X" in axis:
+            axis_bounds["X"] = {
                 "domain": self._dataset.bounds.get_bounds("X").copy(),
                 "region": lon_bounds,
-            },
-            "Y": {
+            }
+        if "Y" in axis:
+            axis_bounds["Y"] = {
                 "domain": self._dataset.bounds.get_bounds("Y").copy(),
                 "region": lat_bounds,
-            },
-        }
+            }
```

Comment:
Previously, the `axis_bounds` dict tried to populate `X` and `Y` every time, but we do not need to populate the bounds information if we are not performing spatial averaging over the `X` or `Y` axis. This fixes #324. I think this was a bug and I didn't see a need to modify the unit tests in this case.

Reply:
Logic looks good. I did some minor refactoring and will push the code for your review.